cvpysdk/datacube.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# --------------------------------------------------------------------------
+# Copyright ©2016 Commvault Systems, Inc.
+# See LICENSE.txt in the project root for
+# license information.
+# ---
+
+"""
+    Main file for performing Datacube related operations
+
+    The class 'Datacube' is defined here
+
+    Datacube:
+    __init__(commcell_object)    --  initialise object of the Datacube class
+    _attribs_()                  --  initializes the objects of the classes given in the input list
+
+    _init_attrib_()              --  initializes the object of the class given as input and stores
+                                        in the given input dictionary with class name as key
+
+"""
+from threading import Thread
+
+try:
+    # Python 2 import
+    from Queue import Queue
+except ImportError:
+    # Python 3 import
+    from queue import Queue
+
+from .exception import SDKException
+from .datasource import Datasources
+
+
+class Datacube(object):
+
+    """ Represents a datacube running on the commcell """
+
+    def __init__(self, commcell_object):
+        """Initialize object of the Datacube class.
+
+            Args:
+                commcell_object (object)  --  instance of the Commcell class
+
+            Returns:
+                object - instance of the Datacube class
+        """
+        self._commcell_object = commcell_object
+
+        # APIs
+        self._ANALYTICS_ENGINES = self._commcell_object._services['GET_ANALYTICS_ENGINES']
+        self._ALL_DATASOURCES = self._commcell_object._services['GET_ALL_DATASOURCES']
+
+        # API results
+        self.analytics_engines = self._get_analytics_engines()
+
+        datacube_sdk_classes = [
```

Comment:
Will other classes come in this list later.
If not, then don't use this part, and queues